src/test/java/com/example/universe/simulator/entityservice/unit/controllers/PlanetControllerTest.java

```diff
@@ -0,0 +1,311 @@
+                .post("/planet/add")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(planetDto));
+        MockHttpServletResponse response = mockMvc
+                .perform(requestBuilder)
+                .andReturn()
+                .getResponse();
+        // assert
+        verifyErrorResponse(response.getContentAsString(), ErrorCodeType.MISSING_PARAMETER_NAME);
+        then(service).should(never()).add(any());
+    }
+
+    @Test
+    void testAdd_validateEmptyName() throws Exception {
+        // arrange
+        PlanetDto planetDto = TestUtils.buildPlanetDtoForAdd();
+        planetDto.setName("");
+        // act
+        RequestBuilder requestBuilder = MockMvcRequestBuilders
+                .post("/planet/add")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(planetDto));
+        MockHttpServletResponse response = mockMvc
+                .perform(requestBuilder)
+                .andReturn()
+                .getResponse();
+        // assert
+        verifyErrorResponse(response.getContentAsString(), ErrorCodeType.MISSING_PARAMETER_NAME);
+        then(service).should(never()).add(any());
+    }
+
+    @Test
+    void testAdd_validateBlankName() throws Exception {
+        // arrange
+        PlanetDto planetDto = TestUtils.buildPlanetDtoForAdd();
+        planetDto.setName(" ");
+        // act
+        RequestBuilder requestBuilder = MockMvcRequestBuilders
+                .post("/planet/add")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(planetDto));
+        MockHttpServletResponse response = mockMvc
+                .perform(requestBuilder)
+                .andReturn()
+                .getResponse();
+        // assert
+        verifyErrorResponse(response.getContentAsString(), ErrorCodeType.MISSING_PARAMETER_NAME);
+        then(service).should(never()).add(any());
+    }
+
+    @Test
+    void testUpdate() throws Exception {
+        // arrange
+        PlanetDto planetDto = TestUtils.buildPlanetDtoForUpdate();
+        Planet planet = modelMapper.map(planetDto, Planet.class);
+        given(service.update(planet)).willReturn(planet);
+        // act
+        RequestBuilder requestBuilder = MockMvcRequestBuilders
+                .put("/planet/update")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(planetDto));
+        MockHttpServletResponse response = mockMvc
+                .perform(requestBuilder)
+                .andReturn()
+                .getResponse();
+        // assert
+        verifySuccessfulResponse(response, planetDto);
```

Comment:
Here you should use modelMapper.map(entity, PlanetDto.class) instead of planetDto. Input and output dtos might be different.